src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -283,27 +336,19 @@ def _get_imports_info(
     address: Address,
     owners_per_import: Iterable[PythonModuleOwners],
     parsed_imports: ParsedPythonImports,
-    explicitly_provided_deps: ExplicitlyProvidedDependencies,
 ) -> dict[str, ImportResolveResult]:
     def _resolve_single_import(owners, import_name) -> ImportResolveResult:
         if owners.unambiguous:
             return ImportResolveResult(ImportOwnerStatus.unambiguous, owners.unambiguous)
 
-        explicitly_provided_deps.maybe_warn_of_ambiguous_dependency_inference(
-            owners.ambiguous,
-            address,
-            import_reference="module",
-            context=f"The target {address} imports `{import_name}`",
-        )
-        maybe_disambiguated = explicitly_provided_deps.disambiguated(owners.ambiguous)
-        if maybe_disambiguated:
-            return ImportResolveResult(ImportOwnerStatus.disambiguated, (maybe_disambiguated,))
-        elif import_name.split(".")[0] in DEFAULT_UNOWNED_DEPENDENCIES:
+        if import_name.split(".")[0] in DEFAULT_UNOWNED_DEPENDENCIES:
             return ImportResolveResult(ImportOwnerStatus.unownable)
         elif parsed_imports[import_name].weak:
-            return ImportResolveResult(ImportOwnerStatus.weak_ignore)
+            # what is weak_ignore? should we try to disambiguate these?
+            return ImportResolveResult(ImportOwnerStatus.weak_ignore, owners.ambiguous)
         else:
-            return ImportResolveResult(ImportOwnerStatus.unowned)
+            # return ImportResolveResult(ImportOwnerStatus.unowned)
+            return ImportResolveResult(ImportOwnerStatus.ambiguous, owners.ambiguous)
```

Comment:
I think it might not be strictly correct to have the else branch mark them all as `ImportOwnerStatus.ambiguous`. I think we can know at this point if there are no owners (ambiguous or otherwise) _and_ that it's not ok that this should be the case (owned by stdlib, weak). We could mark those as `ImportOwnerStatus.unowned` here.
I think it will still work this way as `owners.ambiguous` will be empty and it won't be possible to disambiguate that.